test/mocks/upstream/cluster_info.cc

```diff
@@ -29,6 +29,7 @@ MockClusterInfo::MockClusterInfo()
       resource_manager_(new Upstream::ResourceManagerImpl(runtime_, "fake_key", 1, 1024, 1024, 1)) {
 
   ON_CALL(*this, connectTimeout()).WillByDefault(Return(std::chrono::milliseconds(1)));
+  ON_CALL(*this, idleTimeout()).WillByDefault(Return(idle_timeout_));
```

Comment:
Having the member variable `idle_timeout_` in the mock doesn't do anything since `Return` captures by value. Either use `ReturnPointee` or just `Return` a temporary here.